app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class AccountsController < ApplicationController
+  before_action :authenticate_admin!, only: %i[show index edit destroy]
+  
+  def index
+    @accounts = Account.all
```

Comment:
Better approach to push variables to the view is using `locals` it is like this
```ruby
def index
  render locals: { account: Account.all }
end
``` 
This pattern is safer when you use nesting in the viewses. 
I the view then you are using `account` instead of `@account`
You should apply this to all you controller actions and viewses